packages/server/src/fhir/repo.test.ts

```diff
@@ -461,6 +483,24 @@ describe('FHIR Repo', () => {
     expect(patient2?.meta?.lastUpdated).not.toEqual(lastUpdated);
   });
 
+  test('Update Resource with Missing id', async () => {
+    const patient1 = await systemRepo.createResource<Patient>({
+      resourceType: 'Patient',
+      name: [{ family: 'Test' }],
+    });
+
+    const { id, ...rest } = patient1;
+    expect(id).toBeDefined();
+    expect((rest as Patient).id).toBeUndefined();
+
+    try {
+      await systemRepo.updateResource<Patient>(rest);
+      fail('Should have thrown');
+    } catch (err) {
+      expect((err as OperationOutcomeError).outcome).toMatchObject(badRequest('Missing id'));
```

Comment:
Add coverage for https://sonarcloud.io/code?id=medplum_medplum&pullRequest=1978&selected=medplum_medplum%3Apackages%2Fserver%2Fsrc%2Ffhir%2Frepo.ts&line=512